kvlds/btree.c

```diff
@@ -247,6 +247,12 @@ btree_init(struct wire_requestqueue * Q_lbs, uint64_t npages,
 		goto err1;
 	}
 
+	/* Warn about unnecessarily large page lengths and exit. */
+	if (T->pagelen > 4096) {
```

Comment:
If you're on board such a check in theory, this could be replaced with a formula relating to `keylen` and `vallen`.